### DISCUSSION TITLE
How do I use it with typescript??

### DISCUSSION THREAD

**Discussion by SubigyaLamichhane** [nativewind/nativewind#134] [Q&A]
I read the documentation and as suggested added a my-app.d.ts file with 

`/// <reference types="nativewind/types" />`

but got the following error "Cannot find type definition file for 'nativewind/types'."

**Comment by marklawlor** [comment]
Can you please publish a repository with your code? You can verify that `nativewind/types` should work by checking that `node_modules/nativewind/types.d.ts` exists. If that file is present then there is a problem with your typescript configuration.

**Comment by SubigyaLamichhane** [comment]
This is my ts-config file.
`{
  "name": "react-native-tutorial",
  "version": "1.0.0",
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "dev": "expo start --tunnel",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "eject": "expo eject",
    "gen": "graphql-codegen --config codegen.yml"
  },
  "dependencies": {
    "@apollo/client": "^3.6.9",
    "@react-native-community/hooks": "^2.8.1",
    "@types/crypto-js": "^4.1.1",
    "cloudinary-react": "^1.8.1",
    "crypto-js": "^4.1.1",
    "expo": "~45.0.0",
    "expo-status-bar": "~1.3.0",
    "formik": "^2.2.9",
    "graphql": "^16.5.0",
    "nativewind": "^2.0.0-next.53",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-native": "0.68.2",
    "react-native-cloudinary": "^1.0.1",
    "react-native-gesture-handler": "~2.2.1",
    "react-native-maps": "0.30.2",
    "react-native-reanimated": "~2.8.0",
    "react-native-safe-area-context": "4.2.4",
    "react-native-web": "0.17.7",
    "react-navigation": "^4.4.4",
    "react-navigation-stack": "^2.10.4",
    "react-redux": "^8.0.2",
    "redux": "^4.2.0",
    "redux-thunk": "^2.4.1",
    "tailwind-rn": "^4.2.0",
    "tailwindcss-react-native": "^1.7.10"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@graphql-codegen/cli": "^2.10.0",
    "@graphql-codegen/typescript": "^2.7.2",
    "@graphql-codegen/typescript-operations": "^2.5.2",
    "@graphql-codegen/typescript-react-apollo": "^3.3.2",
    "@types/react": "~17.0.21",
    "@types/react-native": "~0.66.13",
    "tailwindcss": "^3.1.6",
    "typescript": "~4.3.5"
  },
  "private": true
}
`

**Reply by marklawlor** [reply]
@SubigyaLamichhane That is your `package.json` file, not your `tsconfig.json`.

Can you please share your repository as the problem may be your project structure, and not with a single file.

**Comment by SubigyaLamichhane** [comment]
Oh I am sorry. 
Here, is my tsconfig file
`{
  "extends": "expo/tsconfig.base",
  "compilerOptions": {
    "strict": true
  },
  "include": ["./**/*"]
}
`

**Reply by marklawlor** [reply]
Like I said, showing me a single file does not help at all. If you cannot share your repository (or create a new one demonstrating the issue) then I cannot help you further.

**Comment by danstepanov** [comment]
Closing this discussion, if this is still relevant, feel free to open a new one.
